Framework/API/inc/MantidAPI/WorkspaceNearestNeighbours.h

```diff
@@ -12,7 +12,9 @@
 #include <memory>
 // Boost graphing
 #ifndef Q_MOC_RUN
+#define BOOST_ALLOW_DEPRECATED_HEADERS
```

Comment:
Can we define this in CMake instead rather than having to change the source code? We already have some global boost definitions at https://github.com/mantidproject/mantid/blob/d0872d5641a9be426bd5d04fd556e3e61e854e97/buildconfig/CMake/CommonSetup.cmake#L72

Reply:
So for the deprecated headers specifically I wanted to suppress the warning for only the header this #define is encompassing.